### DISCUSSION TITLE
Add runnable actions to issues

### DISCUSSION THREAD

**Discussion by tsvendsen** [ankitpokhrel/jira-cli#461] [Ideas]
To support e.g checking out branches and so on a runnable action could be configurable and any script / program being called could get the jira issue key passed to stdin.

This way we could easily implement git integrations that check out branches based on issues directly from the list view

I currently have scripts that take this key as an input and checkout a branch, which are named by the summary and then it uses this key to do transitions as I pass through the dev cycle. 
At the moment the way to do this is to use issue list, then quit and run my scripts with the key, it would have been super nice to be able to run an action on an issue instead as my workflow could actually start from the issue list, and my further interactions could also go through that list when I wanted to create merge requests and transition issue at the same time.

This would also make it possible to do all other actions like transition/assign.... to any issue from the list by configuring actions

**Comment by ankitpokhrel** [comment]
I think this is somewhat related to #403 but with more control?

**Comment by tsvendsen** [comment]
I agree to some degree, but what I wanted to do was to run an arbitrary
script/program on my machine with the issue as stdin input.
 I'm trying to automate my workflow not just the Jira part.

Br,
Tommy

søn. 11. sep. 2022, 09:17 skrev Ankit Pokhrel ***@***.***>:

**Comment by duckbrain** [comment]
Let's imagine `jira issues list` has a `--return` flag, which makes it so that if I hit enter on a specific issue, it prints out that key to stdout and exits with status code 0; if I hit q it quits with status code 1.

----

If that were the case, I could compose scripts to do all sorts of things, including interacting with source control. 

```bash
#!/bin/bash
# Check out a branch for a particular issue
git checkout -b "feature/$(jira issues list --return --label 'features_to_work_on')"
```

From there, I can start integrating those scripts into my editor.

It'd be extra useful to print out the selected issue [as JSON](https://github.com/ankitpokhrel/jira-cli/discussions/256), so we could check other fields in the scripts
